pkg/pillar/zedUpload/azureutil/azure.go

```diff
@@ -155,6 +158,61 @@ func DownloadAzureBlob(accountName, accountKey, containerName, remoteFile, local
 	return nil
 }
 
+// DownloadAzureBlobByChunks will process the blob download by chunks, i.e., chunks will be
+// responded back on as and hwen they recieve
+func DownloadAzureBlobByChunks(accountName, accountKey, containerName, remoteFile, localFile string, httpClient *http.Client, chunkChan utils.ChunkDataChan) error {
+	chunkData := utils.ChunkData{}
+	c, err := NewClient(accountName, accountKey, httpClient)
+	if err != nil {
+		return err
+	}
+	blobClient := c.GetBlobService()
+	container := blobClient.GetContainerReference(containerName)
+	containerExists, err := container.Exists()
+	if err != nil {
+		return err
+	}
+	if !containerExists {
+		return fmt.Errorf("Container :%s doesn't exist ", containerName)
+	}
+	blob := container.GetBlobReference(remoteFile)
+	exists, err := blob.Exists()
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("%s: blob doesn't exist", remoteFile)
+	}
+	getErr := blob.GetProperties(nil)
+	if getErr != nil {
+		return getErr
+	}
+	chunkSize := SingleMB
+	chunkData.Size = int64(blob.Properties.ContentLength)
+	fmt.Println("chunkData.Size", chunkData.Size)
+	cw := &utils.ChunkWriter{Data: chunkData, ChunkNotifier: chunkChan}
+	readCloser, err := blob.Get(nil)
+	if err != nil {
+		return err
+	}
+	defer readCloser.Close()
+	var processed int64
+	for processed < chunkData.Size {
+		var rbuf bytes.Buffer
+		bufferSize := math.Min(float64(chunkData.Size-processed), float64(chunkSize))
```

Comment:
I already said I think it is a very bad idea to convert its to floats due to loss of accuracy.

What is wrong with
  bufferSize := chunkData.Size-processed
  if bufferSize > chunkSize {
    bufferSize = chunkSize
  }

Reply:
I was math.Min which was is expecting its type to be in float64.
will change as you suggested.